cmd/proxy/main.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/cortexlabs/cortex/pkg/lib/logging"
+	"github.com/cortexlabs/cortex/pkg/proxy"
+)
+
+const (
+	_reportInterval        = 10 * time.Second
+	_requestSampleInterval = 1 * time.Second
+)
+
+func main() {
+	var (
+		port              int
+		userPort          int
+		targetConcurrency int
+		maxConcurrency    int
+	)
+
+	flag.IntVar(&port, "port", 8000, "port where the proxy will be served")
+	flag.IntVar(&userPort, "user-port", 8080, "port where the proxy will redirect to the traffic to")
+	flag.IntVar(&targetConcurrency, "target-concurrency", 0, "target concurrency for user container")
+	flag.IntVar(&maxConcurrency, "max-concurrency", 0, "max concurrency for user container")
```

Comment:
It's not clear to me what these are based on the descriptions, but that's ok, I think we can merge it in for now, since I wanted to discuss this behavior as a team anyways.